src/sql/base/browser/ui/selectBox/selectBox.ts

```diff
@@ -56,8 +71,12 @@ export class SelectBox extends vsSelectBox {
 
 	private element: HTMLElement;
 
+
+
 	constructor(options: string[], selectedOption: string, contextViewProvider: IContextViewProvider, container?: HTMLElement, selectBoxOptions?: ISelectBoxOptions) {
-		super(options.map(option => { return { text: option }; }), 0, contextViewProvider, undefined, selectBoxOptions);
+		//originally {text :option };
+		super(options.map(option => {return {text : SelectBox.parseName(option)}; }), 0, contextViewProvider, undefined, selectBoxOptions);
```

Comment:
This creates a weird cyclic dependency. A component shouldn't know what's going into it - if there are general rules its applying it should be for everything rather than a specific usage of it.

For example, say this was a "Hot Water" component, a "Hot Water" component doesn't need to know its going to be used for "Tea" or "Coffee". It just needs to know how to behave on its own.

Essentially you should be doing this mapping right before it comes in here, rather than on the component itself.

Another thing you could do is extend this class and make a special case class for graph SelectBoxes - but even then if the only behavior that is different is the name, we should just pass that as a property.